### PR TITLE
Upgrade to latest angular-ui-bootstrap version

### PR DESCRIPTION
**Pull request by jjlangholtz**
Upgrade to the most recent version of `angular-ui-bootstrap`. This
includes a significant number of bug fixes and improvements for a
library that we heavily use.

The biggest breaking change for the upgrade is the addition of a `uib`
prefix to all of `angular-ui-bootstrap` directives. This PR updates all
deprecated usage to the new version.

Because another dependency `angular-patternfly` is still using an older
version of `angular-ui-bootstrap`, a few minor things will not work
until the upgrade is complete there (the about modal and some tooltips).

A PR is open to make the changes there so hopefully it will be updated
in the next release https://github.com/patternfly/angular-patternfly/pull/353

https://github.com/angular-ui/bootstrap/wiki/Migration-guide-for-prefixes

@miq-bot add_label euwe/no, technical debt, enhancement

### CODE REVIEW

**Comment by chriskacerguis** [comment]
@AllenBW can you give this a quick review?

File: package.json [hunk 1]
```diff
@@ -133,6 +132,7 @@
     "yargs": "^5.0.0"
   },
   "dependencies": {
+    "angular-ui-bootstrap": "^2.3.0",
```
Comment (AllenBW) [inline package.json]: Dev dependency fo this jazz  `yarn add angular-ui-bootstrap --dev`

**Comment by AllenBW** [comment]
Gonna make a running list of observations of things that this breaks (pending update from ang-patternfly):
* (as was stated in the comments) about modal till patternfly updates
* tooltips on toolbar actions
* toolbar filtering (this one is a biggie)

**Comment by jjlangholtz** [comment]
Good 👀 @AllenBW on the toolbar catches, I just tested the toolbar against https://github.com/patternfly/angular-patternfly/pull/353 and that PR will fix those issues as well

**Comment by AllenBW** [comment]
@jjlangholtz **TEAMWORKFTW** ㊙️  so those things aside operation nominal (comparing to latest master)

**Comment by AllenBW** [comment]
@jjlangholtz can ya **[wip]** this pr (pending aforementioned pr merge)

**Comment by chriskacerguis** [comment]
@jjlangholtz @AllenBW looks like https://github.com/patternfly/angular-patternfly/pull/353 has been merged, can we finish this up?

**Comment by jjlangholtz** [comment]
@chriskacerguis We still need a new release for angular patternfly to be deployed, alternatively we could point `package.json` at master temporarily but it is probably better just to wait.

**Comment by AllenBW** [comment]
ditto on the wait (sadly)

**Comment by dtaylor113** [comment]
Hi, patternfly sprint ends tomorrow, so a new version will most likely be published by cob Wednesday.
- Dave

**Comment by jjlangholtz** [comment]
@chriskacerguis @AllenBW A new version of angular-patternfly was cut today which included the `uib` prefixes. I believe this is ok to merge now.

**Comment by jjlangholtz** [comment]
@miq-bot remove_label wip

**Comment by dtaylor113** [comment]
Hi, angular-patternfly hasn't been released yet, but should be soon.  It will be angular-patternfly 3.15.3.

**Comment by dtaylor113** [comment]
Hi, angular-patternfly hasn't been released yet, latest draft release note says version "3.16.0".
https://github.com/patternfly/angular-patternfly/releases/tag/untagged-3238977ebc80d5685ef2
We are still working on patternfly/rcue release/build issues.
Thanks for your patience!
- Dave

**Comment by dtaylor113** [comment]
Hi,
angular-patternfly v 3.16.0 was just released :smile: 
Thanks,
- Dave

**Comment by jjlangholtz** [comment]
Thanks @dtaylor113, just upgraded to 3.16 and everything seems to be working great!

**Comment by chriskacerguis** [comment]
@jjlangholtz can you fix the conflict?

**Pull request merged**